src/js/components/search/filters/awardAmount/SpecificAwardAmountItem.jsx

```diff
@@ -20,86 +18,106 @@ export default class SpecificAwardAmountItem extends React.Component {
         super(props);
 
         this.state = {
-            min: 0,
-            max: 0,
-            hideCustom: true
+            min: '',
+            max: '',
+            showWarning: false,
+            warningMessage: ''
         };
 
         this.searchSpecificRange = this.searchSpecificRange.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
+        this.minChange = this.minChange.bind(this);
+        this.maxChange = this.maxChange.bind(this);
     }
 
-    componentDidMount() {
-        this.setupInputListeners();
+    onKeyDown(e) {
+        // Enter
+        if (e.keyCode === 13) {
+            e.preventDefault();
+            this.searchSpecificRange();
+        }
     }
 
-    setupInputListeners() {
-        [this.minValue, this.maxValue].forEach((target) => {
-            target.addEventListener('keydown', (e) => {
-                // Enter
-                if (e.keyCode === 13) {
-                    e.preventDefault();
-                    this.searchSpecificRange();
-                }
-            });
-        });
+    minChange(e) {
+        const min = e.target.value;
+        this.setState({ min }, this.verifyNumberLogic);
+    }
+
+    maxChange(e) {
+        const max = e.target.value;
+        this.setState({ max }, this.verifyNumberLogic);
     }
 
     searchSpecificRange() {
-        const min = AwardAmountHelper.ensureInputIsNumeric(this.minValue.value);
-        let max = AwardAmountHelper.ensureInputIsNumeric(this.maxValue.value);
+        const { min, max } = this.state;
+        this.props.searchSpecificRange([min, max]);
+    }
 
-        if (min >= max) {
-            // If minimum is larger than maximum, take minimum value
-            max = 0;
+    verifyNumberLogic() {
+        if (this.state.max === '') {
+            if (this.state.showWarning) {
+                this.setState({ showWarning: false, warningMessage: '' });
+            }
+            return;
+        }
+        const min = Number(this.state.min);
+        const max = Number(this.state.max);
+        let showWarning = false;
+        const warningMessage = 'Please choose a min less than or equal to the max';
+        const minIsNull = (!min && min !== 0);
+        const maxIsNull = (!max && max !== 0);
+        if (min < max) showWarning = false;
+        if (min > max) showWarning = true;
+        if (minIsNull) showWarning = false;
+        if (maxIsNull) showWarning = false;
+        if (showWarning !== this.state.showWarning) {
+            this.setState({ showWarning, warningMessage });
         }
-
-        this.setState({
-            min,
-            max,
-            hideCustom: false
-        });
-
-        this.props.searchSpecificRange([min, max]);
     }
 
     render() {
-        const hide = this.state.hideCustom ? ' hide' : '';
-
+        const onKeyDownHandler = createOnKeyDownHandler(this.onKeyDown);
```

Comment:
This can become `createOnKeyDownHandler(this.searchSpecificRange)` since the helper handles checking the key code